nixos/tests/samba.nix

```diff
@@ -5,34 +5,52 @@ import ./make-test-python.nix ({ pkgs, ... }:
 
   meta.maintainers = [ pkgs.lib.maintainers.eelco ];
 
-  nodes =
-    { client =
-        { pkgs, ... }:
-        { fileSystems = pkgs.lib.mkVMOverride
-            { "/public" = {
-                fsType = "cifs";
-                device = "//server/public";
-                options = [ "guest" ];
-              };
-            };
+  nodes = {
+    client = { pkgs, ... }: {
+      services.samba-wsdd = {
+        enable = true;
+        interface = "eth1";
+        workgroup = "WORKGROUP";
+        hostname = "CLIENT-WSDD";
+        discovery = true;
+        extraOptions = [ "--verbose" "--no-host" ];
+      };
+      fileSystems = pkgs.lib.mkVMOverride {
+        "/public" = {
+          fsType = "cifs";
+          device = "//server/public";
+          options = [ "guest" ];
         };
+      };
+      networking.firewall.allowedTCPPorts = [ 5357 ];
+      networking.firewall.allowedUDPPorts = [ 3702 ];
```

Comment:
We should mention these ports are required for wsdd discovery only. Or maybe entirely put this into another `client_discovery` VM?

Reply:
Can I transfer it to a separate test - samba-wsdd?